pkg/distributors/contracts/MultiDistributor.sol

```diff
@@ -529,7 +529,15 @@ contract MultiDistributor is IMultiDistributor, ReentrancyGuard, MultiDistributo
 
         // We hold stakingTokens in an external balance as BPT needs to be external anyway
         // in the case where a user is exiting the pool after unstaking.
-        stakingToken.safeTransferFrom(from, address(this), amount);
+        IVault.UserBalanceOp[] memory ops = new IVault.UserBalanceOp[](1);
+        ops[0] = IVault.UserBalanceOp({
+            asset: IAsset(address(stakingToken)),
+            amount: amount,
+            sender: from,
+            recipient: payable(address(this)),
+            kind: IVault.UserBalanceOpKind.TRANSFER_EXTERNAL
+        });
+        getVault().manageUserBalance(ops);
```

Comment:
God everytime I see this abomination that we need to put up with in order to create a fixed size array I want to scream until the eardrums of the solc dev team burst

Reply:
But uh, yea, good idea re. transferring assets via the Vault! 

Something to keep in mind however is that this will cause issues with autostaking, as the Pool won't be able to stake during a join because of the Vault reentrancy. Maybe we do need two ways to stake after all if we want to support the autostake scenario.